src/libraries/Microsoft.Bcl.Memory/tests/Microsoft.Bcl.Memory.Tests.csproj

```diff
@@ -24,10 +24,19 @@
     <Compile Include="..\..\System.Memory\tests\Base64Url\Base64UrlValidationUnitTests.cs">
       <Link>System\Memory\Base64Url\Base64UrlValidationUnitTests.cs</Link>
     </Compile>
+    <Compile Include="..\..\System.Runtime\tests\System.Runtime.Tests\System\RangeTests.cs">
+      <Link>System\RangeTests.cs</Link>
+    </Compile>
+    <Compile Include="..\..\System.Runtime\tests\System.Runtime.Tests\System\IndexTests.cs">
+      <Link>System\IndexTests.cs</Link>
```

Comment:
Do these tests validate that the .. and ^ syntax is usable? Without Index/Range, that syntax will fail to compile, so we should just sanity check that by including the Microsoft.Bcl.Memory package, that syntax starts working.

Reply:
Yes, the tests use the new syntax, the only case not supported after this change is `[i..j]` because it needs `RuntimeHelpers.GetSubArray`. That is why I added this method in the test project. We can choose to add it to the package too but I am not sure if this is going to cause any trouble. If we add it, will be added as internal type.

Reply:
> Yes, the tests use the new syntax

Great, thanks.

> We can choose to add it to the package too but I am not sure if this is going to cause any trouble

We can't add it as public as it would conflict with the actual RuntimeHelpers, and adding it as internal won't help, as the C# compiler needs to be able to emit calls to it.

Reply:
It is added as internal in the test, and it works. is it different?

Reply:
We need to think if we can include `RuntimeHelpers.GetSubArray` as  internal tyep later to allow `[i..j]` compiler syntax.

Reply:
> It is added as internal in the test, and it works. is it different?

It'll work for any code in the project, because it has access to it. For it to be useful to a consumer, it would need to be public, so that the C# compiler can emit calls to it.

Reply:
Maybe the compiler team can help with this issue. Either the compiler emits code for `GetSubArray` or fallback to another type we can include in the MS.Bcl.Memory library.